### ISSUE TITLE
Refactor `minesweeper-ui` module's `Composables`

### ISSUE THREAD

**Issue by JayaSuryaT** [JayaSuryaT/minesweeper-j-compose#88]
### There are a bunch of refactors to do, _and mostly all of them are a bit iffy._

- [x] Consider removing the gradient background.
- [x] Use a different model for UI representation of `Cells` and `Grid`.
- [x] Move `StatefulGrid` to `ui-game` module, or at least a different version of it.
- [x] General refactors.

---

### Why consider removing the gradient?

While I personally love how it looks, and it took a _goood_ amount of time to figure out how to achieve and implement that in the very specific way I wanted (A shared gradient as a background across all of the un-revealed cells), there are other things to consider.

It is hindering the UI rendering a _wee bit_. If not for the Gradient, there would be no _"Punch hole cells through which there would be a shared background"_ thing. Every cell would be independent, there would be no need for clipping, or inverse clipping, maybe animations (the currently non-existent ones) could be handled better.

With a more straight forward approach than this, reading the code would make much more sense to any random reader.

And apart from all of this, not having to worry about laying out, inverse clipping and having a little bit of more freedom may lead to something new / better / exciting.

---

### Why use a different model for UI representation of `Cells` and `Grid`?

To layout and show a `Unflagged` `Cell` the knowledge of the underlying `MineCell` is not required, just the information of `Cell`'s position should be enough. But currently this _(unnecessary)_ information is being passed on to the UI layer as well. This is happening as we are using the same model to execute logic and represent UI.

Because of this _unnecessary_ data being present in the _current_ _UI_ model, there were some unnecessary animations being triggered. And because of that issue, `Cell` animations were completely removed at that point, and also the performance was bad.

Things can be a bit better if we could use a separate model for laying out and rendering a `RawCell`. (Albeit there would be a little bit of mapping involved, which might induce some memory consumption, especially considering this mapping would get triggered at every cell update for the entire grid. But at this point everything is speculation, we'll see when we'll get there.)

#### TLDR; Animations are not working properly, and performance is not great; Maybe this will fix it.

---

### Why move `StatefulGrid` to `ui-game` ?

It was pretty obvious by the name, but I see it just now that this model `StatefulGrid` is strictly a UI model, being used for storing UI state. At that point is has no business to be in `minesweeper-engine` module. 

So, it should be moved to a more appropriate place, which would be `ui-game`. 

And while we are at it, the `StatefulGrid` should also be updated for any of the changes required for the `Cell` and `Grid` model updates.
